[Android] Update Kotlin / gradle version in Android Chiptool

joonhaengHeo opened the pull request:
Fix #33836 
After Kotlin compiler was updated to 2.0, gradle version and kotlin compiler version need to be changed to prevent build errors.
Please review whether these modification points are correct.

### Testing
This PR is only related Build Script. Test is not required.

joonhaengHeo committed:
Update Kotlin / gradle version in Android Chiptool

joonhaengHeo commented:
@yunhanw-google 
I would like to fix the gradle version change, including this PR. However, the java version in the validation script is set to 1.8, so I can't change it. Is there a way to make this possible?

joonhaengHeo committed:
Change Java version

joonhaengHeo committed:
Update to Java 11

yunhanw-google commented:
@andreilitvin, could you help bumper docker to validate this change in CI?

andy31415 commented:
> @andreilitvin, could you help bumper docker to validate this change in CI?

@joonhaengHeo please separate out a docker image update PR and we can merge that.

andy31415 commented on integrations/docker/images/stage-2/chip-build-java/Dockerfile:
```diff
@@ -1,12 +1,12 @@
 RUN set -x \
     && apt-get update \
     && DEBIAN_FRONTEND=noninteractive apt-get install -fy \
-    openjdk-8-jdk \
```
changes to dockerfile need to update the version of images in chip-build (because chip-build is the base, we share that version with all. This is easier to manage than trying to have each docker file maintain its own version)

joonhaengHeo commented:
@andy31415 @yunhanw-google 
I uploaded new PR only related Docker. Please check about this.
#37130

joonhaengHeo committed:
Revert Dockerfile Version

joonhaengHeo committed:
Update version

joonhaengHeo committed:
install cmdline tools

joonhaengHeo committed:
Merge branch 'master' into change_android_gradle_kotlin_version

joonhaengHeo commented:
@yunhanw-google 
I tried after updating java 11. But, cmdline-tools can't support java 11. (cmdline-tools is supported in Java 17.)
Please let me know how to solve this.

yunhanw-google commented:
You may need create another PR to update commandline tool?

andy31415 committed:
Merge branch 'master' into change_android_gradle_kotlin_version

andy31415 committed:
Merge branch 'master' into change_android_gradle_kotlin_version

andy31415 committed:
Use 105 image

andy31415 committed:
Update smoketest-android.yaml

yunhanw-google committed:
Update android.py with latest sdk path

yunhanw-google committed:
Revert "Update android.py with latest sdk path"

This reverts commit 864dc0e5a9ffe167f392ca0adb839dc66aa449fb.

yunhanw-google committed:
Fix compilation errors

-- Use docker 106
-- Remove unsupported -XX:MaxPermSize=2048m
-- Update java version

yunhanw-google committed:
Merge branch 'master' into change_android_gradle_kotlin_version

andy31415 committed:
Merge branch 'master' into change_android_gradle_kotlin_version

yunhanw-google committed:
bump classpath gradle to 7.1.3 for other android build

yunhanw-google committed:
update the rest gradle across tress

andy31415 commented:
Fast track: fixes android builds

yunhanw-google committed:
bump class path to 7.3.3 as well

yunhanw-google committed:
Revert "bump class path to 7.3.3 as well"

This reverts commit 1cd60b06756449eec3da15d7c2130519a880124c.

yunhanw-google committed:
Merge branch 'master' into change_android_gradle_kotlin_version
